backend/src/middlewares/auth/auth.middleware.ts

```diff
@@ -32,4 +35,26 @@ const localAuth = (
   )(req, res, next);
 };
 
-export { localAuth };
+const signUpMiddleware = async (
+  req: TypedRequestBody<Prisma.UserCreateInput>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
+  try {
+    await signUpSchema.validate(req.body);
+    const existedUser = await prisma.user.findFirst({
+      where: {
+        email: req.body.email,
+      },
+    });
+    if (existedUser) throw new Error();
```

Comment:
I don`t understand.
If user exists you throw new error?

Reply:
@RomanMahola 
This is the sign **_up_** middleware. So throwing an exception is actually a expected :slightly_frowning_face:. The only thing I would change is to add the message the `Error` constructor

Reply:
Yes, I added error message.